### PR TITLE
Initial server setup

### PR DESCRIPTION
**Pull request by dplumb94**
Signed-off-by: Darian Plumb <dplumb@bitwise.io>

### CODE REVIEW

File: bin/run_lint [hunk 1]
```diff
@@ -32,7 +32,13 @@ lint() {
 
 ret_val=0
 
+
 lint addressing/marketplace_addressing || ret_val=1
+lint marketplace_transaction/transaction_creation || ret_val=1
```
Comment (delventhalz) [inline bin/run_lint]: Can this be linted without adding it to the PYTHONPATH?

File: docker-compose-installed.yaml [hunk 2]
```diff
@@ -44,7 +44,14 @@ services:
       - market-ledger-sync
       - rethink
       - validator
-    command: echo "No REST API command set"
+    command: |
+      bash -c "
+        sleep 1 &&
+        ./bin/rest-api \
```
Comment (delventhalz) [inline docker-compose-installed.yaml]: Let's name this `marketplace-rest-api`

File: docker-compose.yaml [hunk 3]
```diff
@@ -46,7 +46,11 @@ services:
       - market-ledger-sync
       - rethink
       - validator
-    command: echo "No REST API command set"
+    command: |
+      bash -c "
+        sleep 3 &&
```
Comment (delventhalz) [inline docker-compose.yaml:51]: The ideal way to handle this is write it so it gracefully attempts to reconnect if the db or validator are unavailable. I will see if I can add this to a later PR for both ledger-sync and the rest-api.

File: marketplace_transaction/transaction_creation/common.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+
+class Key(object):
```
Comment (delventhalz) [inline marketplace_transaction/transaction_creation/common.py]: I believe this is being removed from Boyd's PR. You may need to rebase, or also remove it here.

File: rest_api/config.py.example [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+DB_NAME = 'marketplace'
+
+# Runtime settings
+DEBUG = True
+
+# Secret keys
+# WARNING! These defaults are insecure, and should be changed for deployment
+SECRET_KEY = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'  # any string
+AES_KEY = 'ffffffffffffffffffffffffffffffff'  # 32 character hex string
+BATCHER_PRIVATE_KEY = '1111111111111111111111111111111111111111111111111111111111111111'  #64 character hex string
```
Comment (delventhalz) [inline rest_api/config.py.example]: Add a space after `#` . . . it will bug me.

**Review by delventhalz — CHANGES_REQUESTED** [review]

File: rest_api/Dockerfile [hunk 6]
```diff
@@ -33,4 +34,4 @@ WORKDIR /project/sawtooth-marketplace
 
 EXPOSE 8000/tcp
 
-CMD ["echo \"No server command set\""]
+CMD ["./bin/rest-api"]
```
Comment (boydjohnson) [inline rest_api/Dockerfile]: It looks like the bin is called `marketplace-rest-api`.

**Review by boydjohnson — APPROVED** [review]

**Review by delventhalz — APPROVED** [review]

**Pull request merged**